### PR TITLE
release: announce v0.11.0

### PR DESCRIPTION
**Pull request by benesch**
Polish off the release notes for v0.11.0, plus update the docs to announce our new beta support for prebuilt ARM binaries.

### CODE REVIEW

**Review by philip-stoev — APPROVED** [review]
As we now have ARM in the CI, I am OK with the wording re ARM support. We will need to do another review before declaring it GA.

File: doc/user/content/install.md [hunk 1]
```diff
@@ -11,12 +11,16 @@ Want to connect with Materialize? Join our growing community on Slack! →
...
 install on macOS and Linux, or [build](#build-from-source) on most OSes (e.g. FreeBSD). These
-instructions install the latest release of Materialize, **{{< version >}}**. The latest (unstable)
-developer builds are available at https://dev.materialize.com/. For prior releases,
-see the [Versions page](/versions).
+instructions install the latest release of Materialize, **{{< version >}}**. For prior releases and unstable builds, see the [Versions page](/versions).
 
 **Have any questions?** [Ask us on Slack](https://materialize.com/s/chat)
 
+{{< warning >}}
+Support for the ARM CPU architecture is in beta. You may encounter performance
```
Comment (morsapaes) [inline doc/user/content/install.md:19]:
```suggestion
Support for the ARM64 CPU architecture is in beta. You may encounter performance
```

File: doc/user/content/release-notes.md [hunk 2]
```diff
@@ -107,42 +107,57 @@ List new features before bug fixes.
...
 {{% version-header v0.11.0 %}}
 
-- Accept message names in [Protobuf sources] that do not start with a leading
-  dot. This fixes a regression introduced in v0.9.12.
+- **Breaking change.** Remove the `mz_workers` function {{% gh 9363 %}}.
 
-- Queries that are too big to fit the limits of our intermediate representations
-  no longer panic and will now cause query-level failure with an internal error
-  and a message of the form "exceeded recursion limit of {X}".
+- Provide [pre-compiled binaries](/versions) for the ARM64 CPU architecture.
```
Comment (morsapaes) [inline doc/user/content/release-notes.md:112]:
```suggestion
- Provide [pre-compiled binaries](/versions) for the ARM CPU architecture.
```

Comment (morsapaes) [reply doc/user/content/release-notes.md:112]: (Very pedantic) nit, just so we mention either ARM or AMR64 consistently.

**Review by morsapaes — APPROVED** [review]

**Pull request merged**